lib/active_record_shards/shard_selection.rb

```diff
@@ -1,23 +1,17 @@
 # frozen_string_literal: true
 module ActiveRecordShards
   class ShardSelection
-    NO_SHARD = :_no_shard
-    cattr_accessor :default_shard
-
-    def initialize
+    def initialize(shard)
       @on_slave = false
-      @shard = nil
+      self.shard = shard
     end
 
     if ActiveRecord::VERSION::MAJOR < 5
 
       def shard(klass = nil)
-        if (@shard || self.class.default_shard) && (klass.nil? || klass.is_sharded?)
-          if @shard == NO_SHARD
-            nil
-          else
-            @shard || self.class.default_shard
-          end
+        if klass.nil? || klass.is_sharded?
```

Comment:
If we don’t like `klass` being `nil`, perhaps we should change the method signature?

Reply:
No wait, this is Rails 4 code. Let’s just leave it…